src/ray/util/logging.cc

```diff
@@ -335,11 +341,22 @@ std::string RayLog::GetLogFormatPattern() { return log_format_pattern_; }
 
 std::string RayLog::GetLoggerName() { return logger_name_; }
 
+void RayLog::SetExposeLogCallback(const ExposeLogCallback &expose_log_callback) {
+  expose_log_callback_ = expose_log_callback;
+}
+
 RayLog::RayLog(const char *file_name, int line_number, RayLogLevel severity)
-    : logging_provider_(nullptr), is_enabled_(severity >= severity_threshold_) {
+    : logging_provider_(nullptr),
+      is_enabled_(severity >= severity_threshold_),
+      severity_(severity),
+      is_exposed_(severity == RayLogLevel::FATAL) {
+  if (is_exposed_) {
+    expose_osstream_ = std::make_shared<std::ostringstream>();
+    *expose_osstream_ << file_name << ":" << line_number << ":";
+  }
   if (is_enabled_) {
-    logging_provider_ =
-        new LoggingProvider(file_name, line_number, GetMappedSeverity(severity));
+    logging_provider_ = new LoggingProvider(
```

Comment:
Is there any reason why we do raw malloc instead of using smart pointer for this btw?

Reply:
Maybe because we want to delete this provider to flush log in an special time? I don't really know.